### ISSUE TITLE
[Bug] Uncaught Error: Invalid period label: day

### ISSUE THREAD

**Issue by mneudert** [piwik/piwik#22191]
### What happened?

**tldr:** An active Marketplace plugin can break the period logic, causing lots of issues.

Viewing any reporting page, and then changing the date using the period selector, you can find an uncaught error in the JavaScript console:

```
index.php?module=Proxy&action=getUmdJs&chunk=2

Uncaught Error: Invalid period label: day
    at e.value (index.php?module=Proxy&action=getUmdJs&chunk=2)
    at e.value (index.php?module=Proxy&action=getUmdJs&chunk=2)
    at Ie.updateDateInTitle (index.php?module=Proxy&action=getUmdJs&chunk=2)
    at e.value (index.php?module=Proxy&action=getUmdJs&chunk=0)
    at index.php?module=Proxy&action=getUmdJs&chunk=0
```

This error can lead to a broken behaviour, for example the home dashboard not updating (showing the old date despite the change in the period selector).

The error is making only partial sense, as the main page load was already successful, hinting at the periods being available and initialized. But then, for the page update, the periods somehow vanished.

Deactivating the JS merging (`disable_merged_assets = 1`) in development mode gave a more meaningful error:

```
Marketplace.umd.min.js?cb=5115:7

Uncaught Error: Invalid period label: month
    at e.value (Marketplace.umd.min.js?cb=5115:7:446)
    at e.value (Marketplace.umd.min.js?cb=5115:7:548)
    at Ie.updateDateInTitle (Marketplace.umd.min.js?cb=5115:13:247)
    at e.value (CoreHome.umd.min.js?cb=5115:76:3097)
    at CoreHome.umd.min.js?cb=5115:76:947
```

Both the `CoreHome.umd.js` and `Marketplace.umd.js` have a call to initialize the period storage:

```
/* harmony default export */ var Periods_Periods = (new Periods());
```

But the  Marketplace does not have the individual period objects in the build, no calls to `Periods_Periods.addCustomPeriod` to be found there.

To confirm this assumption I deactivated the Marketplace plugin and the error disappeared. After activing the plugin again, it reappeared.

### What should happen?

Changing the date in the period selector should always work.

### How can this be reproduced?

Change the selected date on the dashboard with an activated Marketplace plugin.

An uncaught error should appear in the JavaScript console, and the dashboard not update properly. This is easily visible with a "Visits Over Time" widget on the dashboard, the dates displayed won't change after the dashboard reloaded.

### Matomo version

5.1.x

### What browsers are you seeing the problem on?

Firefox, Chrome

### Validations

- [X] Read our [Contributing Guidelines](https://github.com/matomo-org/matomo/blob/5.x-dev/CONTRIBUTING.md).
- [X] Follow our [Security Policy](https://github.com/matomo-org/matomo/blob/5.x-dev/SECURITY.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating duplicates.
- [X] The provided steps to reproduce is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the Bug.
